From 1d0f70af70bc8b3cee615a083c945ff6dada039e Mon Sep 17 00:00:00 2001
From: "Yue, Liang" <liang.yue@cienet.com.cn>
Date: Mon, 24 Jul 2017 09:11:21 +0800
Subject: [PATCH 1/3] base change

1.recovery: if install error still restart
2.wifi: fix .ko insmod error
3.notification along with inter active timeout.
---
 .../java/com/android/server/power/PowerManagerService.java    | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/frameworks/base/services/core/java/com/android/server/power/PowerManagerService.java b/frameworks/base/services/core/java/com/android/server/power/PowerManagerService.java
index 2697ae4..b6a1cf5 100755
--- a/frameworks/base/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/frameworks/base/services/core/java/com/android/server/power/PowerManagerService.java
@@ -103,6 +103,7 @@ public final class PowerManagerService extends SystemService
     private static final int MSG_SANDMAN = 2;
     // Message: Sent when the screen brightness boost expires.
     private static final int MSG_SCREEN_BRIGHTNESS_BOOST_TIMEOUT = 3;
+    private static final int MSG_NOTIFY_ACTIVITY_TIMEOUT = 4;
 
     // Dirty bit: mWakeLocks changed
     private static final int DIRTY_WAKE_LOCKS = 1 << 0;
@@ -614,6 +615,7 @@ public final class PowerManagerService extends SystemService
             updateSettingsLocked();
             mDirty |= DIRTY_BATTERY_STATE;
             updatePowerStateLocked();
+            mHandler.sendEmptyMessageDelayed(MSG_NOTIFY_ACTIVITY_TIMEOUT, 30 * 1000);
         }
     }
 
@@ -1007,6 +1009,9 @@ public final class PowerManagerService extends SystemService
         synchronized (mLock) {
             if (userActivityNoUpdateLocked(eventTime, event, flags, uid)) {
                 updatePowerStateLocked();
+                Slog.d(TAG, "user active, start compute timeout");
+                mHandler.removeMessages(MSG_NOTIFY_ACTIVITY_TIMEOUT);
+                mHandler.sendEmptyMessageDelayed(MSG_NOTIFY_ACTIVITY_TIMEOUT, 30 * 1000);
             }
         }
     }
@@ -2833,6 +2838,12 @@ public final class PowerManagerService extends SystemService
                 case MSG_SCREEN_BRIGHTNESS_BOOST_TIMEOUT:
                     handleScreenBrightnessBoostTimeout();
                     break;
+                case MSG_NOTIFY_ACTIVITY_TIMEOUT: {
+                    Slog.d(TAG,"MSG_NOTIFY_ACTIVITY_TIMEOUT");
+                    Intent intent = new Intent("com.example.lock.NOTIFY_ACTIVITY_TIMEOUT");
+                    mContext.sendBroadcast(intent);
+                }
+                    break;
             }
         }
     }
-- 
1.9.1

